packages/core/src/actions/execute.ts

```diff
@@ -750,28 +1006,241 @@ export const runEntireDeploymentProcess = async (
+ * to improve the chance that bugs will get caught during the initial proposal simulation step rather than occuring
+ * when a deployment is actually happening on the website.
+ *
+ * The allow for this function to be reused in all three situations this function accepts a `deploymentContext` object
+ * which implements all the logic that needs to be different in each of the three situations.
+ *
+ * We aim to *only* use the `deploymentContext` for logic which absolutely cannot be shared. For example, logic for
+ * writing information about a deployment to the website backend is implemented in the `deploymentContext`.
+ *
+ * See documentation on the `DeploymentContext` type at the top of this file for more information of the specific fields
+ * and justification for why the logic implemented in them cannot be shared.
+ */
+export const compileAndExecuteDeployment = async (
+  deploymentContext: DeploymentContext
+): Promise<
+  | {
+      receipts: SphinxTransactionReceipt[]
+      batches: SphinxLeafWithProof[][]
+      finalStatus: BigInt
+      failureAction: HumanReadableAction | undefined
+    }
+  | undefined
+> => {
+  const { deployment } = deploymentContext
+  const networkName = deployment.networkName
+  const deploymentId = deployment.id
+  const rawCompilerConfigs = deployment.compilerConfigs
+  const { logger } = deploymentContext
+
+  if (!rawCompilerConfigs) {
+    deploymentContext.throwError(
+      `[Executor ${networkName}]: No compiler config stored for this deployment, should never happen`,
+      deployment.id,
+      networkName
+    )
+  }
+
+  logger?.info(`[Executor ${networkName}]: retrieving the deployment...`)
+  // Compile the bundle using either the provided localDeploymentId (when running the in-process
+  // executor), or using the Config URI
+  let merkleTree: SphinxMerkleTree
+  let compilerConfigs: Array<CompilerConfig>
+  let configArtifacts: ConfigArtifacts
+
+  // Handle if the config cannot be fetched
+  try {
+    ;({ merkleTree, compilerConfigs, configArtifacts } =
+      await buildDeploymentWithCompilerConfigs(rawCompilerConfigs!))
+  } catch (e: any) {
+    await deploymentContext.handleError(e, deployment, networkName, [])
+
+    return
+  }
+
+  const targetNetworkConfig = compilerConfigs.find(
+    (config) => config.chainId === deployment.chainId.toString()
+  )
+  if (!targetNetworkConfig) {
+    await deploymentContext.throwError(
+      `[Executor ${networkName}]: Error could not find target network config. This should never happen please report it to the developers.`,
+      deploymentId,
+      networkName
+    )
+    return
+  }
+
+  const { projectName } = targetNetworkConfig.newConfig
+
+  // get active deployment ID for this project
+  const moduleAddress = deployment.moduleAddress
+  const module = new ethers.Contract(
+    moduleAddress,
+    SphinxModuleABI,
+    deploymentContext.provider
+  )
+
+  if ((await deploymentContext.provider.getCode(moduleAddress)) !== '0x') {
```

Comment:
We do attempt to verify in `handleAlreadyExecutedDeployment`. You could probably make an argument that we should refactor this so that we don't have to call `verify` within this function, but I think that would require a somewhat involved refactor. I would probably wait until we do the ticket to share the verification logic to improve this.